autoscaling.tf

```
@@ -24,40 +24,58 @@ resource "aws_autoscaling_policy" "scale_down" {
-  count               = local.autoscaling_enabled ? 1 : 0
-  alarm_name          = "${module.this.id}${module.this.delimiter}cpu${module.this.delimiter}utilization${module.this.delimiter}high"
-  comparison_operator = "GreaterThanOrEqualToThreshold"
-  evaluation_periods  = var.cpu_utilization_high_evaluation_periods
-  metric_name         = "CPUUtilization"
-  namespace           = "AWS/EC2"
-  period              = var.cpu_utilization_high_period_seconds
-  statistic           = var.cpu_utilization_high_statistic
-  threshold           = var.cpu_utilization_high_threshold_percent
 
-  dimensions = {
-    AutoScalingGroupName = join("", aws_autoscaling_group.default.*.name)
+locals {
+  default_alarms = {
+    cpu_high = {
+      alarm_name          = "${module.this.id}${module.this.delimiter}cpu${module.this.delimiter}utilization${module.this.delimiter}high"
+      comparison_operator = "GreaterThanOrEqualToThreshold"
+      evaluation_periods  = var.cpu_utilization_high_evaluation_periods
+      metric_name         = "CPUUtilization"
+      namespace           = "AWS/EC2"
+      period              = var.cpu_utilization_high_period_seconds
+      statistic           = var.cpu_utilization_high_statistic
+      threshold           = var.cpu_utilization_high_threshold_percent
+      dimensions_name     = "AutoScalingGroupName"
+      dimensions_target   = join("", aws_autoscaling_group.default.*.name)
+      alarm_description   = "Scale up if CPU utilization is above ${var.cpu_utilization_high_threshold_percent} for ${var.cpu_utilization_high_period_seconds} seconds"
+      alarm_actions       = [join("", aws_autoscaling_policy.scale_up.*.arn)]
+    },
+    cpu_low = {
+      alarm_name          = "${module.this.id}${module.this.delimiter}cpu${module.this.delimiter}utilization${module.this.delimiter}low"
+      comparison_operator = "LessThanOrEqualToThreshold"
+      evaluation_periods  = var.cpu_utilization_low_evaluation_periods
+      metric_name         = "CPUUtilization"
+      namespace           = "AWS/EC2"
+      period              = var.cpu_utilization_low_period_seconds
+      statistic           = var.cpu_utilization_low_statistic
+      threshold           = var.cpu_utilization_low_threshold_percent
+      dimensions_name     = "AutoScalingGroupName"
+      dimensions_target   = join("", aws_autoscaling_group.default.*.name)
+      alarm_description   = "Scale down if the CPU utilization is below ${var.cpu_utilization_low_threshold_percent} for ${var.cpu_utilization_low_period_seconds} seconds"
+      alarm_actions       = [join("", aws_autoscaling_policy.scale_down.*.arn)]
+    }
   }
-
-  alarm_description = "Scale up if CPU utilization is above ${var.cpu_utilization_high_threshold_percent} for ${var.cpu_utilization_high_period_seconds} seconds"
-  alarm_actions     = [join("", aws_autoscaling_policy.scale_up.*.arn)]
+  all_alarms = merge(var.custom_alarms, local.default_alarms)
```

```suggestion
  all_alarms = merge(local.default_alarms, var.custom_alarms)
```

Let's allow the callers to override the default alarms